src/reformat.c

```diff
@@ -1082,7 +1082,10 @@ avifResult avifImageYUVToRGB(const avifImage * image, avifRGBImage * rgb)
     }
 
     avifBool convertedWithLibYUV = AVIF_FALSE;
-    avifResult libyuvResult = avifImageYUVToRGBLibYUV(image, rgb);
+    avifResult libyuvResult = AVIF_RESULT_NOT_IMPLEMENTED;
```

Comment:
I suggest two changes.

1. Move the if-else statement for `libyuvResult == AVIF_RESULT_OK` into the new if statement you added.
2. Move the declaration of the `alphaMultiplyMode` local variable earlier and use it in the new if statement.


Here is the patch:

```
diff --git a/src/reformat.c b/src/reformat.c
index 68b25ea..acbbfda 100644
--- a/src/reformat.c
+++ b/src/reformat.c
@@ -1078,18 +1078,20 @@ avifResult avifImageYUVToRGB(const avifImage * image, avifRGBImage * rgb)
         return AVIF_RESULT_REFORMAT_FAILED;
     }
 
+    avifAlphaMultiplyMode alphaMultiplyMode = state.toRGBAlphaMode;
     avifBool convertedWithLibYUV = AVIF_FALSE;
-    avifResult libyuvResult = avifImageYUVToRGBLibYUV(image, rgb);
-    if (libyuvResult == AVIF_RESULT_OK) {
-        convertedWithLibYUV = AVIF_TRUE;
-    } else {
-        if (libyuvResult != AVIF_RESULT_NOT_IMPLEMENTED) {
-            return libyuvResult;
+    if (alphaMultiplyMode == AVIF_ALPHA_MULTIPLY_MODE_NO_OP || avifRGBFormatHasAlpha(rgb->format)) {
+        avifResult libyuvResult = avifImageYUVToRGBLibYUV(image, rgb);
+        if (libyuvResult == AVIF_RESULT_OK) {
+            convertedWithLibYUV = AVIF_TRUE;
+        } else {
+            if (libyuvResult != AVIF_RESULT_NOT_IMPLEMENTED) {
+                return libyuvResult;
+            }
         }
     }
 
     // Reformat alpha, if user asks for it, or (un)multiply processing needs it.
-    avifAlphaMultiplyMode alphaMultiplyMode = state.toRGBAlphaMode;
     if (avifRGBFormatHasAlpha(rgb->format) && (!rgb->ignoreAlpha || (alphaMultiplyMode != AVIF_ALPHA_MULTIPLY_MODE_NO_OP))) {
         avifAlphaParams params;
 
```

Reply:
Done.
Changes to `src/reformat_libyuv.c` are now removed from this PR. I'll include it in the planned PR to update libyuv usage,  when libyuv's 16 bit support is finished.